### PR TITLE
Add loading pill delay

### LINKED ISSUE #3501: Only show Loading... pill if the request is taking more than X seconds (0.5s <= X <= 1s)
Requested by Sam.

X to be determined

### PR DESCRIPTION
**Pull request by diegovar**
## Summary

This introduces a 1s delay before the loading pill is shown, as requested by UX.

## Relevant Technical Choices

setTimeout and some state to determine whether it must be shown.

## User-facing changes

Loading pill now takes 1s before appearing. If loading takes less than that, it doesn't appear at all.

## Testing Instructions

Verify that browsing works fine in the gallery, with the caveat that the loading pill takes longer to appear.

---

Fixes #3501

### CODE REVIEW

File: assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js [hunk 1]
```diff
@@ -167,7 +169,20 @@ function PaginatedMediaGallery({
       </div>
     );
 
-  const displayLoadingPill = isMediaLoading && hasMore;
+  const [showLoadingPill, setShowLoadingPill] = useState(false);
+
+  useEffect(() => {
+    if (isMediaLoading) {
```
Comment (joannag6) [inline assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js]: Can this just be if (isMediaLoading && hasMore)? Then we can combine `displayLoadingPill` and `showLoadingPill`

File: assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js [hunk 2]
```diff
@@ -167,7 +169,20 @@ function PaginatedMediaGallery({
...
+      const showLoadingTimeout = setTimeout(() => {
+        setShowLoadingPill(isMediaLoading);
+      }, SHOW_LOADING_PILL_DELAY_MS);
+      return () => clearTimeout(showLoadingTimeout);
+    }
+    setShowLoadingPill(false);
+    return undefined;
```
Comment (joannag6) [inline assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js:182]: Is this necessary?

**Review by joannag6 — APPROVED** [review]

File: assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js (hunk 2 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js:182]: Unfortunately it is. If you return sth from an effect (to be executed on unmount), you have yto be consistent and always return sth, undefined in the case that you want nothing to occur in that situation.

File: assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js (hunk 1 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js]: Done

File: assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js (hunk 2 above)
Comment (joannag6) [reply assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js:182]: Ah, is `return undefined` preferred over `return () => {}`?

Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js:182]: Yes, the linter explicitly said to return undefined in that case.

**Pull request merged**